x-pack/plugins/monitoring/public/components/apm/apm_metrics.tsx

```diff
@@ -44,6 +44,15 @@ const createCharts = (series: unknown[], props: Partial<Props>) => {
 
 export const ApmMetrics = ({ stats, metrics, seriesToShow, title, ...props }: Props) => {
   const topSeries = [metrics.apm_cpu, metrics.apm_memory, metrics.apm_os_load];
+  const inAgent = stats.config?.container && stats.config?.agentMode;
```

Comment:
If users have a dedicated monitoring deployment, then this check would be true/false for the whole monitoring cluster, but the clusters shipping monitoring data might not all run an Elastic Agent.

From 7.13 on, ECE/ESS will always start an Elastic Agent and have an APM & Fleet Server running. Therefore what we need to check is 
* version of the monitored deployment  >= 7.13 (@igoristic is it possible to query for the version here?)
* is it a ECE/ESS deployment (afaik this information should be available from the cloud plugin)

If we can figure these two things out we wouldn't even need an additional config settings IMO.

Reply:
Yeah, that would be great, if we don't have to use the config. Doesn't look like the cloud plugin has that available, but I'm trying to figure out if there are any other alternatives

Reply:
@simitt 
> is it a ECE/ESS deployment

There isn't an official API to do this yet, but it's already being discussed here: https://github.com/elastic/kibana/issues/93416. The problem still remains that we might not know the environment of the cluster that's being **monitored**, since we don't have that in the `.monitoring-es*` indices.

> version of the monitored deployment >= 7.13

We can easily detect this for the local cluster, and the monitored/data cluster since we have the `version` field in `cluster_stats` in `.monitoring-es*` indices

---

I agree the config option isn't ideal UX, but not sure if there are any other options

Reply:
Thanks for the analysis - so the version of the monitored cluster can be retrieved, but not whether or not it is a cloud deployment. 

The stack monitoring is showing the same data in both cases, this is really just used for showing a different title. In one scenario it shows the usage from APM Server only, in the second scenario from APM Server & Fleet Server (& Elastic Agent). Can we find a name that covers both scenarios? @nehaduggal and @formgeist could you get involved here? For keeping things simple, it would be very helpful if we found a better naming, and avoiding to have to make this distinction in the code. 

Screen in question:
<img width="1765" alt="Screenshot 2021-03-25 at 10 50 10" src="https://user-images.githubusercontent.com/5555349/112453689-39b90f80-8d58-11eb-8798-cdf930e0983d.png">

Reply:
Because of lack of better ideas, I'm inclined on moving forward with this by reusing how we detect whether or not to show `cgroup` values, combined with the deployment version of the monitored cluster. So basically changing the Title to `APM & Fleet Server` in the overview section and to `APM & Fleet Server - Resource Usage`,  whenever the monitoring cluster is a cloud deployment and the monitored cluster is >= `7.13`. 
The kind of data that are shown are not changing, it is only the title that is displayed. 

@nehaduggal @elastic/apm-server what do you think?

Reply:
Thank you, for concluding the UI. This actually makes it easier to implement (and is a better flow/experience imo), since it's no long a predetermined config option